src/providers/one_inch.rs

```diff
@@ -59,6 +67,79 @@ impl OneInchProvider {
             .send()
             .await
     }
+
+    async fn get_token_price(
+        &self,
+        chain_id: &str,
+        address: &str,
+        currency: &PriceCurrencies,
+    ) -> Result<String, RpcError> {
+        let base = format!("{}/price/v1.1/{}/{}", &self.base_api_url, chain_id, address);
+        let mut url = Url::parse(&base).map_err(|_| RpcError::ConversionParseURLError)?;
```

Comment:
Using `join`:

```suggestion
        let mut url = self.base_api_url.join(format!("/price/v1.1/{}/{}", chain_id, address)).map_err(|_| RpcError::ConversionParseURLError)?;
```

Also are we properly validating/URL encoding `chain_id` and `address`? I'd default to URL encoding in-case something upstream forgets to validate.

Reply:
Thanks for the good comments! Join is much nicer! 
In case of bad encoding (if someone will use other than utf8 in 2024 😅) there will be an error that the chain_id/address is in the wrong format with the bad characters in the error, so the error should be obvious if it happens.

Reply:
I meant if their chain ID or address has a slash in it and creates an unexpected URL

Reply:
Yes, this makes sense. We need to check this earlier. Thanks for catching this.

Reply:
I've added the proper format validation for the chain ID and address during the CAIP-2/10 disassembling in utils. This will cut such type of URL injection attack where CAIP2/10 disassembling is used.